src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerServer.cs

```diff
@@ -94,15 +94,21 @@ public void Run()
                             break;
 
                         case InstallRequestType.WriteInstallStateFile:
-                            WriteInstallStateFile(new SdkFeatureBand(request.SdkFeatureBand), request.InstallStateContents);
+                            WriteInstallStateFile(request.DotnetDir, new SdkFeatureBand(request.SdkFeatureBand), request.InstallStateContents);
                             Dispatcher.ReplySuccess($"Created install state file for {request.SdkFeatureBand}.");
                             break;
 
                         case InstallRequestType.RemoveInstallStateFile:
-                            RemoveInstallStateFile(new SdkFeatureBand(request.SdkFeatureBand));
+                            RemoveInstallStateFile(request.DotnetDir, new SdkFeatureBand(request.SdkFeatureBand));
                             Dispatcher.ReplySuccess($"Deleted install state file for {request.SdkFeatureBand}.");
                             break;
 
+                        case InstallRequestType.AdjustWorkloadMode:
+                            AdjustInstallMode(request.DotnetDir, new SdkFeatureBand(request.SdkFeatureBand), request.ProductCode);
+                            string newMode = request.ProductCode.Equals("true", StringComparison.OrdinalIgnoreCase) ? "workload sets" : "loose manifests";
```

Comment:
The PR description mentions 3 possible values: workloadset, loosemanifest, and auto.  It looks like this only handles the first two, and represents the value as a string.

Could we represent the value as an Enum, or possibly a nullable Boolean?

Reply:
This doesn't handle 'auto' because auto is already handled in the WorkloadUpdateCommand where it is the same as loosemanifest (for now) but will later be changed to being the same as workloadset. Since there are only two possible values at this stage, I'm not sure I see the value of changing it to an enum.